nullaway/src/main/java/com/uber/nullaway/qual/EnsuresNonNull.java

```diff
@@ -0,0 +1,19 @@
+package com.uber.nullaway.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Can annotate a methods with @EnsuresNonnull(param) annotation where param is one of the classes
+ * fields. It indicates a post-condition for the method, that at every call site to this method, the
+ * class field in the argument is @Nonnull at exit point. If a method is annotated
+ * with @EnsuresNonnull(param), NullAway checks weather the @Nonnull assumption of the field at exit
+ * point is valid.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+public @interface EnsuresNonNull {
+  String[] value();
+}
```

Comment:
@msridhar Separate from this PR. Do we have an issue for creating a `com.uber.nullaway:annotation` or `com.uber.nullaway:lib` maven artifact? Would be good to have these out there, together with a canonical implementation of `castToNonNull`, etc.